[linux-6.6.y] ata: ahci: Add support for AHCI SGPIO Enclosure Management

leoliu-oc opened the pull request:
zhaoxin inclusion
category: feature

-------------------

To monitor and control auxiliary service in a drive enclosure, Zhaoxin AHCI controller adds enclosure management support in SGPIO protocols with two messages types:
LED message type and SGPIO register interface message type.

The LED message type uses a genernal ahci specific interface which has already been supported by default ahci driver, the SGPIO register interface message type based on SFF-8485 which defined by vendor specific, this patch adds support for it.

opsiff commented on drivers/ata/Kconfig:
```diff
@@ -115,6 +115,16 @@ config SATA_AHCI
+config AHCI_ZHAOXIN_SGPIO
+	tristate "zhaoxin AHCI SGPIO support"
+	depends on SATA_AHCI
+	default y
```
default y的话 非X86平台也需要这个么?

leoliu-oc replied:
这边在新版本中添加了对CPU_SUP_ZHAOXIN/CPU_SUP_CENTAUR的依赖。这样就只会在兆芯平台上builtin了。

leoliu-oc committed:
ata: ahci: Add support for AHCI SGPIO Enclosure Management

zhaoxin inclusion
category: feature

-------------------

To monitor and control auxiliary service in a drive enclosure, Zhaoxin
AHCI controller adds enclosure management support in SGPIO protocols with
two messages types:
LED message type and SGPIO register interface message type.

The LED message type uses a genernal ahci specific interface which has
already been supported by default ahci driver, the SGPIO register
interface message type based on SFF-8485 which defined by vendor specific,
this patch adds support for it.

Signed-off-by: leoliu-oc <leoliu-oc@zhaoxin.com>

opsiff commented on drivers/ata/ahci_zhaoxin_sgpio.c:
```diff
@@ -0,0 +1,706 @@
+		return 0;
+
+	while (pdev_next) {
+		pdev_next = pci_get_device(PCI_VENDOR_ID_ZHAOXIN, 0x9083, pdev_cur);
```
这种用法没见过，我的理解是PCI上没有这个设备就不注册

opsiff commented:
/ok-to-test
